refactor: fix various "problems" reported by GoLand

erikgb opened the pull request:
## Description

This fixes some "problems" reported by my IDE, GoLand - using the default analyze profile. Motivation: Avoid warnings when I commit/push changes to this project.

I have tried to take the least resistance approach, but feel free to suggest other ways of solving the issues:
- variable name conflicting with imported package
- unhandled errors
- variable names using snake_case

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

josedonizetti commented on pkg/plugins/trivy/plugin.go:
```diff
@@ -1496,8 +1495,8 @@ func (p *plugin) parseImageRef(imageRef string) (v1alpha1.Registry, v1alpha1.Art
 
-	for name, cvss := range CVSSs {
-		if name == "nvd" {
+	for k, cvss := range CVSSs {
```
can we return this one to `name, cvss`?

josedonizetti commented:
@erikgb  This looks good! Only one place I prefer the old version. WDYT?

chen-keinan commented:
@erikgb good initiative , but I'm not sure this pr is good idea got our contributors, as most of our community users/contributors  do not use intellij ide. (some use but not follow warning)  and it will be difficult to keep the code in no warning standard, I prefer not to make life hard for contributors as this point. we have linter for the important stuff.

erikgb commented:
> @erikgb good initiative , but I'm not sure this pr is good idea got our contributors, as most of our community users/contributors do not use intellij ide. (some use but not follow warning) and it will be difficult to keep the code in no warning standard, I prefer not to make life hard for contributors as this point. we have linter for the important stuff.

@chen-keinan Ok, fair enough. I can split it up, then. Symbol names colliding with imported packages is the worst. It is wrong IMO and makes things behave strangely. I'll start with that.

JAORMX commented:
@erikgb you could check for these using golangci-lint, mind enabling the relevant checks in the config so we don't fall into these again?

erikgb commented:
> @erikgb you could check for these using golangci-lint, mind enabling the relevant checks in the config so we don't fall into these again?

@JAORMX Thanks! I didn't know rules were excluded! I'll do!

erikgb commented:
After https://github.com/aquasecurity/trivy-operator/pull/322 was merged, there is not a lot left here, just small fixes of various things. Any suggestions how to proceed? Should I just drop this PR, split it up or try to get it merged? @josedonizetti @chen-keinan

chen-keinan commented:
I think that are good thing here , let`s add it as a one time effort , have you adjust the linter config  (at least for symbols)?

erikgb commented:
> I think that are good thing here , let`s add it as a one time effort , have you adjust the linter config (at least for symbols)?

I will add a linter for the symbols; use camelCase.

erikgb commented:
I've now added the [errcheck](https://github.com/kisielk/errcheck) linter, and added a TODO to add the [nosnakecase](https://github.com/sivchari/nosnakecase) linter when available in the next release of golangci-lint.

erikgb commented:
> I've now added the [errcheck](https://github.com/kisielk/errcheck) linter, and added a TODO to add the [nosnakecase](https://github.com/sivchari/nosnakecase) linter when available in the next release of golangci-lint.

After working with https://github.com/aquasecurity/trivy-operator/pull/339, I don't think there is much to do with the lint-configuration in this PR. I suggest to create an issue to enable the [nosnakecase](https://github.com/sivchari/nosnakecase) linter when available in the next release of golangci-lint. WDYT @chen-keinan?

chen-keinan commented:
ok , then you can close this PR

erikgb commented:
Close it, oh well...... 😞
